ext/geos_c_impl/geometry.c

```diff
@@ -1142,8 +1183,13 @@ method_geometry_make_valid(VALUE self)
   if (!self_geom)
     return Qnil;
 
+  GEOSMakeValidParams* params = GEOSMakeValidParams_create();
+  rgeo_make_valid_apply_params(argc, argv, params);
```

Comment:
This method must not raise in order to keep our garbages properly collected. Or it can raise (`rb_funcall` for instance) or `rgeo_make_valid_check_method` which can raise. So in the current state of the code you have a potential leak. You have to secure this bit of code. In the current state I suggest using rb_protect directly.


See https://ulysse.md/2023/01/24/Calling-Ruby-Methods-in-C-Avoid-Memory-Leaks.html (`rb_protect`) or https://blog.peterzhu.ca/ruby-c-ext-part-8/ (`rb_ensure`)

Reply:
Agree that we should secure this, but what would we need to free in this case? Seems like not much has been done by this point and a failure in `rgeo_make_valid_check_method` already frees the params.

Reply:
We need to free the params! We we call `scan_args` or `rb_funcall`, these are also subjects to a jump unfortunately (and I guess `rb_get_kwargs` could jump as well).

Reply:
i changed it to ruby kwarg processing to avoid c complexity. but not sure if it fits your high standards 🫣

Reply:
I'm sorry you're feeling it like this... This C code can be really tricky, and easily lead to memory leaks. I seems like the new way you are doing it should work, let's see what memcheck has to say. If that works that way I feel like we should make the `geometry_make_valid` private to make sure one is not calling it with wrong arguments. WDYT?

Reply:
sure, i've applied requested change. thx for review